luigi/task.py

```diff
@@ -550,6 +551,37 @@ def on_success(self):
         Default behavior is to send an None value"""
         pass
 
+    @contextmanager
+    def _without_unpicklable_properties(self):
+        """
+        Remove unpicklable properties before dump task and resume them after.
+
+        This method could be called in subtask's dump method, to ensure unpicklable
+        properties won't break dump.
+
+        This method is a context-manager which can be called as below:
+
+        .. code-block: python
+
+            class DummyTask(luigi):
+
+                def _dump(self):
+                    with self._without_unpicklable_properties():
+                        pickle.dumps(self)
+
+        """
+        unpicklable_properties = ('set_tracking_url', 'set_status_message')
+        reserved_properties = {}
+        for property_name in unpicklable_properties:
+            if hasattr(self, property_name):
+                reserved_properties[property_name] = getattr(self, property_name)
+                setattr(self, property_name, None)
```

Comment:
Here is an opportunity to save somebody from great debugging frustration I think. What if instead of `None` we have a human-readable value like `placeholder_during_pickling`. That is so easy to `grep` to.

What do you think?